### PR TITLE
doc(Delta): more documentation about Delta

### PR DESCRIPTION
**Pull request by CatHood0**
## Description

Added more documentation for Delta, works, and examples about it.

- [ ] ✨ **New feature:** Adds new functionality without breaking existing features.
- [ ] 🛠️ **Bug fix:** Resolves an issue without altering current behavior.
- [ ] 🧹 **Code refactor:** Code restructuring that does not affect behavior.
- [ ] ❌ **Breaking change:** Alters existing functionality and requires updates.
- [ ] 🧪 **Tests:** Adds new tests or modifies existing tests.
- [x] 📝 **Documentation:** Updates or additions to documentation.
- [ ] 🗑️ **Chore:** Routine tasks, or maintenance.
- [ ] ✅ **Build configuration change:** Changes to build or deploy processes.

### CODE REVIEW

File: doc/delta_introduction.md [hunk 1]
```diff
@@ -0,0 +1,252 @@
...
+`Delta` is a structured format used to represent text editing operations consistently and efficiently. It is especially useful in collaborative editors where multiple users may be editing the same document simultaneously.
+
+## How does Delta work?
+
+`Delta` consists of a list of operations. Each operation describes a change in the document's content. The operations can be of three types: insertion (`insert`), deletion (`delete`), and retention (`retain`). These operations are combined to describe any change in the document's state.
+
+You can import `Delta` and `Operation` class using:
+
+```dart
+import 'package:flutter_quill/dart_quill_delta.dart';
```
Comment (EchoEllet) [inline doc/delta_introduction.md:12]: This is `quill_delta` instead of `dart_quill_delta`

Comment (CatHood0) [reply doc/delta_introduction.md:12]: I'll fix it. Thanks!

Comment (CatHood0) [reply doc/delta_introduction.md:12]: I already checked this, and i always need to import `dart_quill_delta` to get `Delta` and `Operation` classes. Why do you suggest this?

Comment (EchoEllet) [reply doc/delta_introduction.md:12]: So developers don't have to add `dart_quill_delta` explicitly, we can export additional required files or export `dart_quill_delta`

**Pull request merged**